### DISCUSSION TITLE
Custom icons for close, remove, submenus, and more

### DISCUSSION THREAD

**Discussion by joshbader** [palantir/blueprint#5876] [Q&A] [answered]
Is there a way to provide custom icons to elements such as:

- Dialog/Drawer/Toast: dismiss button
- Tags: remove icon
- Menu Items: submenu caret indicator
- Numerical Inputs: incremental buttons
- etc.

**Comment by adidahiya** [comment, marked as answer]
Generally, no. You may use the CSS APIs to render components that use Blueprint styles (and replace icons with your customizations), but you won't get the interactive functionality provided by the standard JS APIs. The icons are deeply embedded into the core library, and decoupling them is not a design goal at this moment.

- [Dialog CSS API](https://blueprintjs.com/docs/#core/components/dialog.css)
- [Tags CSS API](https://blueprintjs.com/docs/#core/components/tag.css)
- [Menu CSS API](https://blueprintjs.com/docs/#core/components/menu.css)
- Numeric input: N/A
